tutorials/tutorial-01.qmd

```diff
@@ -6,33 +6,1403 @@ time: "~45 minutes"
+#| label: filter-protected-area-layers
+available_layers |>
+  filter(grepl("protected|natura|cdda", layer_name, ignore.case = TRUE)) |>
+  select(layer_name, title)
+```
+
+We'll use three complementary datasets to get comprehensive coverage of marine protected areas:
+
+- **`cddaareas`**: the CDDA database captures nationally designated sites, giving us country-level protected areas
+- **`natura2000areas`**: the EU's flagship conservation network, essential for any European marine analysis
+- **`marineprotectedareas`**: includes sites designated under regional conventions (OSPAR, HELCOM) that may not appear in the other datasets
+
+```{r}
+#| label: define-protected-area-layers
+mpa_layers <- c(
+  "cddaareas",
+  "natura2000areas",
+  "marineprotectedareas"
+)
+```
+
+#### Energy Infrastructure Layers
+
+Now let's find layers related to energy infrastructure. The service includes both power cables (`pcables*`) and telecommunication cables, but we'll focus on energy sector infrastructure:
+
+```{r}
+#| label: filter-infrastructure-layers
+available_layers |>
+  filter(grepl("platform|pipeline|cable", layer_name, ignore.case = TRUE)) |>
+  select(layer_name, title)
+```
+
+For our energy sector focus, we'll use:
+
+- **`platforms`**: offshore installations (oil, gas, wind) as point locations
+- **`pipelines`**: subsea pipelines for oil and gas transport as line features
+- **`pcables*`**: power cables from multiple national sources (BSH, SHOM, Rijkswaterstaat, Norwegian) for offshore wind and interconnector coverage
+
+```{r}
+#| label: define-infrastructure-layers
+infrastructure_layers <- c(
+  "platforms",
+  "pipelines",
+  "pcablesbshcontis",
+  "pcablesshom",
+  "pcablesrijks",
+  "pcablesnve"
+)
+```
+
+### Understanding Layer Structure
+
+WFS layers can contain thousands of features with dozens of attributes. Downloading everything and filtering afterwards wastes bandwidth and time. Before committing to a download, we should:
+
+1. **Discover available attributes**: what columns exist and what types are they?
+2. **Identify filter candidates**: which attributes contain categorical values we can filter on?
+3. **Check actual values**: what values exist so we can write accurate filter expressions?
+
+The `emodnet.wfs` package provides lightweight metadata functions that answer these questions without downloading full datasets. This "look before you leap" approach helps us write efficient, targeted queries.
+
+#### Inspecting Attribute Metadata
+
+The `layer_attribute_descriptions()` function returns metadata about each attribute in a layer: name, type, and whether it's the geometry column. It takes a WFS client connection (`wfs`) and a layer name (`layer`).
+
+Since we have three layers to inspect, we use `map()` to apply the function to each layer name in `mpa_layers`:
+
+```{r}
+#| label: inspect-mpa-attributes
+#| cache: true
+mpa_descriptions <- map(
+  set_names(mpa_layers),
+  ~ layer_attribute_descriptions(wfs = wfs, layer = .x)
```

Comment:
```suggestion
  \(x) layer_attribute_descriptions(wfs = wfs, layer = x)
```

The use of a formula is no longer recommended. https://purrr.tidyverse.org/reference/map.html#arg--f

Should we add a minimal R version somewhere?